src/App.tsx

```diff
@@ -1,62 +1,116 @@
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { getUserPosts } from './api/posts';
+import { User } from './types/User';
+import { Post } from './types/Post';
 
 export const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [loaded, setLoaded] = useState(false);
+  const [hasError, setError] = useState(false);
+
+  const [author, setAuthor] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+
+  function loadUserPosts(userId: number) {
+    setLoaded(false);
+
+    getUserPosts(userId)
+      .then(setPosts)
+      .catch(() => setError(true))
+      .finally(() => setLoaded(true));
+  }
+
+  useEffect(() => {
+    setSelectedPost(null);
+
+    if (author) {
+      loadUserPosts(author.id);
+    } else {
+      setPosts([]);
+    }
+  }, [author?.id]);
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector value={author} onChange={setAuthor} />
               </div>
 
-              <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+              <div
+                className="block"
+                data-cy="MainContent"
+              >
+                {!author && (
+                  <p
+                    data-cy="NoSelectedUser"
+                  >
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {author && !loaded && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {author && loaded && hasError && (
+                  <div
+                    className="notification is-danger"
+                    data-cy="PostsLoadingError"
+                  >
+                    Something went wrong!
+                  </div>
+                )}
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                {author && loaded && !hasError && posts.length === 0 && (
+                  <div
+                    className="notification is-warning"
+                    data-cy="NoPostsYet"
+                  >
+                    No posts yet
+                  </div>
+                )}
 
-                <PostsList />
+                {author && loaded && !hasError && posts.length > 0 && (
```

Comment:
Move common condition into variable (76 and 85 lines) and use this variable
```suggestion
const isPostsShown = author && loaded && !hasError;

....
                { isPostsShown && posts.length && (
```